sw/device/lib/testing/mock_mmio.h

```diff
@@ -7,14 +7,112 @@
 
 #include <stdint.h>
 
+#include <initializer_list>
 #include <memory>
+#include <random>
 #include <vector>
 
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
 #include "sw/device/lib/base/mmio.h"
 
 namespace mock_mmio {
+/**
+ * Represents a single bit field in an integer, useable with EXPECT_* macros
+ * defined in this file.
+ *
+ * An integer can be expressed as a list of BitField values, and can be more
+ * convenient to use than 0b or 0x literals in some cases. For example, the
+ * integer 0b0000'0000'1100'0101 could be expressed as
+ *   {{0x0, 1}, {0x2, 1}, {0x4, 12}}
+ * This form makes it clearer to the reader that 0x0, 0x2, and 04 are indices
```

Comment:
I would use use the `FIELD_FOO_OFFSET, 1` examples here, because this form isn't really that much clearer with literals like this.  The offset usage definitely makes it clear though.

Reply:
My intention was to show what actual values do, and then recommend the OFFSET/MASK usage further down. What do you think?